hazelcast/src/main/java/com/hazelcast/osgi/impl/Activator.java

```diff
@@ -45,17 +45,28 @@ public void start(BundleContext context)
         // Try to start javax.scripting - JSR 223
         activateJavaxScripting(context);
 
-        if (Boolean.getBoolean(HAZELCAST_OSGI_START)) {
-            hazelcastInstance = Hazelcast.newHazelcastInstance();
+        // TODO Is it possible that OSGI service is not null (deactivated) on start?
+        if (hazelcastOSGiService != null) {
+            // If there is an active Hazelcast OSGI service leftover, deactivate it.
+            hazelcastOSGiService.deactivate();
+            hazelcastOSGiService = null;
         }
```

Comment:
I will remove this block as I mentioned in my previous comment and will throw `IllegalStateException`